cardano-node/test/Test/Cardano/Node/FilePermissions.hs

```diff
@@ -0,0 +1,87 @@
+{-# LANGUAGE OverloadedStrings #-}
+{-# LANGUAGE TemplateHaskell #-}
+
+module Test.Cardano.Node.FilePermissions
+  ( tests
+  ) where
+
+import           Cardano.Prelude
+
+import           Control.Exception (bracket_)
+import           System.Directory (removeFile)
+import           System.Posix.Files
+import           System.Posix.IO (createFile)
+import           System.Posix.Types (FileMode)
+
+import           Cardano.Node.Run (checkVRFFilePermissions)
+import           Cardano.Node.Types (VRFPrivateKeyFilePermissionError (..))
+import           Hedgehog (Property, classify, discover, forAll, property, success)
+import qualified Hedgehog
+import           Hedgehog.Internal.Property (failWith)
+
+import           Test.Cardano.Node.Gen (genGroupPermissions, genOtherPermissions)
+
+-- | This property ensures that 'checkVRFFilePermissions' sets the
+-- file permissions & ownership correctly.
+prop_sanityCheck_checkVRFFilePermissions :: Property
+prop_sanityCheck_checkVRFFilePermissions =
+  property $ do
+    -- Correct case: only owner has read permission
+    let correctPermission = ownerReadMode
+        vrfPrivateKeyCorrect = "vrf-private-key-correct"
+    correctResult <-
+      liftIO $ bracket_ (createFile vrfPrivateKeyCorrect correctPermission)
+                        (removeFile vrfPrivateKeyCorrect)
+                        (liftIO . runExceptT $ checkVRFFilePermissions $ Just vrfPrivateKeyCorrect)
+    case correctResult of
+      Left err ->
+        failWith Nothing $ "checkVRFFilePermissions should not have failed with error: "
+                         <> show err
+      Right () -> success
+
+    -- Error case: owner has read permissions & various combinations of other permissions
+    let vrfPrivateKeyOther = "vrf-private-key-other"
+    oPermissions <- forAll genOtherPermissions
+    classify "VRF File has one other permission" $ length oPermissions == 1
+    classify "VRF File has two other permissions" $ length oPermissions == 2
+    classify "VRF File has three other permissions" $ length oPermissions == 3
+    otherResult <-
+      liftIO $ bracket_ (do _ <- createFile vrfPrivateKeyOther nullFileMode
```

Comment:
For some reason I have to create a file with no permissions then set the other permissions. If I try to create the file with other permissions in one go, the file ends up only having owner permissions :man_shrugging: .

Reply:
Weird.

Reply:
Can we leave a brief comment explaining that in the code?